test/unit/math/prim/mat/fun/matrix_exp_multiply_test.cpp

```diff
@@ -29,6 +29,11 @@ inline void test_matrix_exp_multiply() {
 }
 
 TEST(MathMatrix, matrix_exp_multiply) {
+  // the helper above doesn't handle 0 size inputs
+  Eigen::MatrixXd A(0, 0);
+  Eigen::MatrixXd B(0, 0);
+  EXPECT_EQ(stan::math::matrix_exp_multiply(A, B), A);
```

Comment:
Can you add a test for A(0,1) and B(1,2)?

Reply:
In this case we should throw because A is not square. I'm not sure this is what you wanted to see, but I'll add this test anyway.